x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportGetCheckpointAction.java

```diff
@@ -106,7 +106,7 @@ private Map<String, Set<ShardId>> resolveIndicesToPrimaryShards(ClusterState sta
             }
             if (shard.assignedToNode() && nodes.get(shard.currentNodeId()) != null) {
                 // special case: a node that holds the shard is on an old version
-                if (nodes.get(shard.currentNodeId()).getVersion().before(Version.V_8_2_0)) {
+                if (TransformConfigVersion.fromNode(nodes.get(shard.currentNodeId())).before(TransformConfigVersion.V_8_2_0)) {
```

Comment:
This check should be using `TransportVersion`, because it relates to a transport action, not a config.

Since checking `TransportVersion` for a specific node is not possible, the check will have to be simply on the oldest `TransportVersion` in the cluster.